docs/concepts/api-extension/custom-resources.md

```diff
@@ -42,12 +43,162 @@ The [Operator](https://coreos.com/blog/introducing-operators.html) pattern is on
+*   You want to do rolling updates via Deployment, etc, when the file is updated.
+
+Use a secret, which is similar to a configMap, but more secure, for sensitive data.
+
+Use a custom resource (CRD or Aggregated API) if you want do most of the following:
+
+
+
+*   You want to use Kubernetes client libraries and CLIs to create and update the new resource.
+*   You want top-level support from kubectl (e.g. `kubectl get my-object object-name`)
+*   You want to build new automation that watches for updates on the new object, and then CRUD other objects, or vice versa.
+*   You want to write automation that handles updates to the object.
+*   You want to use Kubernetes API conventions like having .spec and .status and .metadata.
+*   You want to object to be an abstraction over a collection of controlled resources, or summarization of other resources.
+
+
+
+## Two Ways to Add Custom Resources
+
+Kubernetes provides two ways to add custom resources to your cluster:
+
+*   [Custom Resource Definitions](https://kubernetes.io/docs/concepts/api-extension/custom-resources/) (CRDs) are easier to use: they do not require any programming in some cases.  
+*   [API Aggregation](https://kubernetes.io/docs/concepts/api-extension/apiserver-aggregation/) requires programming, but allows more control over API behaviors like how data is stored, and conversion between API versions.
+
+Kubernetes provides these two options to meet the needs of different users, so that neither ease-of-use nor flexibility is compromised.
+
+Aggregated APIs are subordinate APIServers that sit behind the primary API server, which acts as a proxy.  This arrangement is called [API Aggregation (AA)](docs/concepts/api-extension/apiserver-aggregation.md).  To users, it simply appears that the Kubernetes API is extended.
+
+Custom Resource Definitions (CRDS) allow users to create new types of resources without adding another APIserver.  You do not need to understand API Aggregation to use CRDs.
+  
+Regardless of whether they are installed via CRDs or AA, the new resources are called Custom Resources to distinguish them from built-in Kubernetes resources (like pods)
+
 ## CustomResourceDefinitions
 
-[CustomResourceDefinition](/docs/tasks/access-kubernetes-api/extend-api-custom-resource-definitions/)
-(CRD) is a built-in API that offers a simple way to create custom resources.
-Deploying a CRD into the cluster causes the Kubernetes API server to begin serving the specified
-custom resource on your behalf.
+
+The [CustomResourceDefinition](/docs/tasks/access-kubernetes-api/extend-api-custom-resource-definitions/) (CRD) resource is a built-in resource that allows defining Custom Resources.  Defining a CRD object creates an new custom resource, with a name and schema that you specify.  The main Kubernetes API server serves and handles the storage of your Custom Resource.
```

Comment:
use /docs/concepts/api-extension/custom-resources/ instead of a task in the link?

Reply:
I'm OK with linking to the task here, but it's a good idea to link to this URL much earlier in the page.